### PR TITLE
chore(stdlib): Add examples to the Array module

### PR DESCRIPTION
**Pull request by sanathusk**
Updated the grains docs for array.gr to add examples for length,get,set,append,concat methods for array

### CODE REVIEW

File: stdlib/array.gr [hunk 1]
```diff
@@ -48,6 +48,7 @@ let initLength = length => {
  * @param array: The array to inspect
  * @returns The number of elements in the array
  *
+ * @example let array = [> 10, 20, 30, 40, 50]; print(Array.length(array)); // prints 5
```
Comment (phated) [inline stdlib/array.gr:51]:
```suggestion
 * @example Array.length([> 1, 2, 3, 4, 5]) == 5
```
Let's shorten these when possible.

File: stdlib/array.gr [hunk 2]
```diff
@@ -145,6 +146,7 @@ export let rec init /*: (Number, Number -> a) -> Array<a>*/ =
  * @param index: The index to access
  * @param array: The array to access
  * @returns The element from the array
+ * @example let array = [> 10, 20, 30, 40, 50]; print(Array.get(1,array)) // prints 20
```
Comment (phated) [inline stdlib/array.gr:149]:
```suggestion
 * @example Array.get([> 1, 2, 3, 4, 5]) == 2
```

File: stdlib/array.gr [hunk 3]
```diff
@@ -162,6 +164,7 @@ export let get = (index, array) => {
  * @param index: The index to update
  * @param value: The value to store
  * @param array: The array to update
+ * @example let array = [> 10, 20, 30, 40, 50]; Array.set(1,22,array); print(array) // prints [> 10, 22, 30, 40, 50]
```
Comment (phated) [inline stdlib/array.gr:167]:
```suggestion
 * @example Array.set(1, 9, [> 1, 2, 3, 4, 5]) == [> 1, 9, 3, 4, 5]
```

File: stdlib/array.gr [hunk 4]
```diff
@@ -177,6 +180,8 @@ export let set = (index, value, array) => {
  * @param array1: The array containing elements to appear first
  * @param array2: The array containing elements to appear second
  * @returns The new array containing elements from `array1` followed by elements from `array2`
+ * @example let array1 = [> 10, 20]; let array2 =[>30, 40, 50]; let combined = Array.append(array1,array2); print(combined) // prints [> 10, 20, 30, 40, 50]
+ // prints [> 10, 22, 30, 40, 50]
```
Comment (phated) [inline stdlib/array.gr:184]:
```suggestion
 * @example Array.append([> 1, 2], [> 3, 4, 5]) == [> 1, 2, 3, 4, 5]
```

File: stdlib/array.gr [hunk 5]
```diff
@@ -199,6 +204,7 @@ export let append = (array1, array2) => {
  *
  * @param arrays: A list containing all arrays to combine
  * @returns The new array
+ * @example let array1 = [>10, 20]; let array2 =[>30, 40, 50]; let combined = Array.concat([array1,array2]); print(combined) // prints [> 10, 20, 30, 40, 50]
```
Comment (phated) [inline stdlib/array.gr:207]:
```suggestion
 * @example Array.concat([[> 1, 2], [> 3, 4], [> 5, 6]]) == [> 1, 2, 3, 4, 5, 6]
```

**Review by phated — CHANGES_REQUESTED** [review]
Thanks for submitting these! I think we should shorten them a bit. We can use `==` to indicate the result instead of using `print` and saying what is printed via a comment.

**Comment by sanathusk** [comment]
@phated  totally agree with your suggestion, pushed  the recommended suggestion.

File: stdlib/array.gr [hunk 6]
```diff
@@ -145,6 +146,7 @@ export let rec init /*: (Number, Number -> a) -> Array<a>*/ =
  * @param index: The index to access
  * @param array: The array to access
  * @returns The element from the array
+ * @example Array.get([> 1, 2, 3, 4, 5]) == 2
```
Comment (phated) [inline stdlib/array.gr:149]: I think I messed up this suggestion. I think it needs 1 as the first argument

**Review by phated — COMMENTED** [review]
@sanathusk I made a mistake in one example

File: stdlib/array.gr (hunk 6 above)
Comment (sanathusk) [reply stdlib/array.gr]: @phated  Done. made the suggested changes

Comment (phated) [reply stdlib/array.gr]: Thanks! I'll kick off CI when I'm at my computer. Can you also make sure you regenerate the markdown docs so CI doesn't fail?

Comment (sanathusk) [reply stdlib/array.gr]: I tried generating markdown and verifying it for the array section by running the command locally:
``` grain doc  stdlib/array.gr --current-version 0.5.3 > array.md ```
Is there any other better way of verifying the correctness md file?

Comment (phated) [reply stdlib/array.gr]: I think that command should work. Have you committed the changes?

Comment (sanathusk) [reply stdlib/array.gr]: yes, committed & pushed

Comment (phated) [reply stdlib/array.gr]: I think you want to overwrite the stdlib/array.md file because you haven't pushed any markdown files in this PR

Comment (sanathusk) [reply stdlib/array.gr]: Should we push the mardown file as well? i have array.md file did not existing in the repo.I generated it by running the above command.I don't think it has to be push, i feel, its generated dynamically.

**Comment by phated** [comment]
@sanathusk We check in the markdown generated docs. Please run `grain doc stdlib -o stdlib --current-version=$(grain -v)` and commit the changed file.

**Comment by sanathusk** [comment]
> grain doc stdlib -o stdlib --current-version=$(grain -v)

@phated  Done . pushed the array.md file as well after running the command `grain doc stdlib -o stdlib --current-version=$(grain -v)`

**Comment by phated** [comment]
Thanks! Running CI now 🎉

**Review by phated — APPROVED** [review]
LGTM! Thanks 🎉

**Review by ospencer — APPROVED** [review]
Fantastic, thanks so much @sanathusk!

**Comment by sanathusk** [comment]
@ospencer @phated  looks like the for the changes to reflect in website the .md files in (https://github.com/grain-lang/grain-lang.org)  needs to be updated. Is the update of .md files in the repo done regularly or is it only during version release.

**Pull request merged**